packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -192,31 +194,41 @@ class ReactShallowRenderer {
 class Updater {
   constructor(renderer) {
     this._renderer = renderer;
+    this._callback = null;
+    this._publicInstance = null;
+  }
+
+  _updateCallback(callback, publicInstance) {
+    if (typeof callback === 'function') {
+      this._callback = callback;
+      this._publicInstance = publicInstance;
+    }
+  }
+
+  _invokeCallback() {
+    if (typeof this._callback === 'function' && this._publicInstance) {
+      this._callback.call(this._publicInstance);
```

Comment:
You'll probably want to immediately set it to null before calling. Since now it is outdated.